pytorch_forecasting/metrics/tests/test_all_metrics.py

```diff
@@ -0,0 +1,135 @@
+            assert isinstance(
+                quantile_pred, torch.Tensor
+            ), "Quantile prediction should be a tensor."  # noqa: E501
+            if y_pred.ndim == 3 or isinstance(metric, PoissonLoss):
+                assert quantile_pred.shape == (
+                    y_pred.shape[0],
+                    y_pred.shape[1],
+                    len(quantiles),
+                ), f"`to_quantiles()`, fails for metric {metric}()."
+            elif y_pred.ndim == 2:
+                assert quantile_pred.shape == (
+                    y_pred.shape[0],
+                    y_pred.shape[1],
+                    1,
+                ), f"`to_quantiles()`, fails for metric {metric}()."
+
+            # testing composite metrics
+            composite = metric + metric
+            weighted = metric * 0.5
+
+            composite_result = composite(y_pred, y)
+            weighted_result = weighted(y_pred, y)
+
+            assert isinstance(composite_result, torch.Tensor)
+            assert isinstance(weighted_result, torch.Tensor)
+            assert torch.isfinite(composite_result).all()
+            assert torch.isfinite(weighted_result).all()
+
+    def _test_reduction_modes(self, metric_class, y_pred, y, test_params):
+        """Test that all metrics support different reduction modes."""
+
+        for reduction in ["mean", "sqrt-mean", "none"]:
+            metric = metric_class(**test_params)
+            if metric_class.__name__ == "NormalDistributionLoss":
+                metric._transformation = None
+            metric.update(y_pred, y)
+            result = metric.reduce_loss(
+                metric.losses, metric.lengths, reduction=reduction
+            )  # noqa: E501
+            assert isinstance(result, torch.Tensor)
+
+            if reduction == "none":
+                result.shape == metric.losses
+            else:
+                assert result.ndim == 0
+                if reduction == "sqrt-mean":
+                    assert (
+                        result >= 0
+                    ), "Result should be non-negative for sqrt-mean reduction."  # noqa: E501
+
+    @pytest.mark.parametrize("target_type", ["standard", "packed", "weighted"])
+    def test_metric_functionality(self, object_pkg, object_class, request, target_type):
+        """Test metric functionality with appropriate test data."""
+        # Skip abstract classes
+        if object_pkg.name in [
+            "Metric",
+            "MultiHorizonMetric",
+            "DistributionLoss",
+        ]:
+            pytest.skip(f"Skipping abstract class {object_pkg.name}")
+
+        prepare_data_fixture_name = object_pkg.requires_data_type()
+        data = request.getfixturevalue(prepare_data_fixture_name)
+
+        test_case = data["test_cases"][target_type]  # noqa: E501
+
+        y_pred, y = object_pkg.prepare_test_inputs(test_case)
+
+        test_params = object_pkg.get_test_params()
+
+        metric = object_class(**test_params)
```

Comment:
this looks the same as `object_instance`

Reply:
can you replace by `object_instance`?

`get_test_params` should return a list of `dict`, not a single `dict`. Then `object_instance` will loop through all these.